### PR TITLE
Updated NewBill Activity UI

### LINKED ISSUE #1: Improve the UI of NewBill Activity
**Is your feature request related to a problem? Please describe.**
Currently, in the [NewBillActivity](../blob/master/markdown_resources/NewBillActivity.jpeg), when you create a bill with the amount in the orders of 10<sup>8</sup> or if the customer's name/address is too long, both these fields overlaps causing bad visual experience.

**Describe the solution you'd like**
Revamp the UI such that this problem is resolved. Any other improvements (like making it more cleaner) in this UI is also welcome.

**Describe alternatives you've considered**
N. A.

**Additional context**
This is the issue I'm trying to point out:

### PR DESCRIPTION
**Pull request by Shashwat545**
## Summary - New Feature/Bug Resolution
Updated the NewBill Activity UI so as to prevent the overlapping of TextViews and implemented a function which preprocesses the amount and displays it in exponential form if it is too large.

## Root Cause
The amount text was not being preprocessed and was being displaying as it was.

## Details
- The width of the Textviews is now restricted to 170 dp. <br>
- The amount text is now being preprocessed and if it is more than 8 digits then it is displayed in exponential form.
- Attached a screenshot for reference below
Screenshot QuickBills

## Checklist
- [x] I've read the [Contribution Guidelines](/CONTRIBUTING.md).
- [x] I've tested that the application is performing well and there is no breaking changes.

## Related Issues / Pull Requests
- Resolves #1

### CODE REVIEW

**Review by sushant-iit — CHANGES_REQUESTED** [review]
There are few modifcations which I want you to consider:
1. The logic behind preprocessing should go in preprocessing function. By this I mean:
if else
This if else block should go to preprocess function, if there is no change just return original value, that's it. This will enable the function to be reused anywhere.
2. Don't hardcode rupee symbol by removing line number 193 from NewBill Activity. This might face difficulty in some devices. Rather use that getString(R.string.rupee, totalAmount) functinon to get rupee symbol. 
3. Merge your changes as a single commit rather than three commits.
Thanks

File: app/src/main/java/com/sushant/quickbills/activity/NewBillActivity.kt [hunk 1]
```diff
@@ -190,7 +191,14 @@ class NewBillActivity : AppCompatActivity(), RecyclerParticularsAdapter.OnClickL
                 )
                 particularList.add(newParticularItem)
                 totalAmount += amount
-                bill_amount.text = getString(R.string.rupee, totalAmount)
+                if(totalAmount>99999999)
```
Comment (sushant-iit) [inline app/src/main/java/com/sushant/quickbills/activity/NewBillActivity.kt]: Put this in preprocess Function

File: app/src/main/java/com/sushant/quickbills/utils/preprocessDisplayText.kt [hunk 2]
```diff
@@ -0,0 +1,8 @@
+package com.sushant.quickbills.utils
+
+fun preprocessDisplayText(num: Double):String
+{
+    val str=num.toBigDecimal().toPlainString()
+    val anss=str.substring(0,1)+"."+str.substring(1,3)+"E"+(str.length-1).toBigDecimal().toPlainString()
```
Comment (sushant-iit) [inline app/src/main/java/com/sushant/quickbills/utils/preprocessDisplayText.kt]: Add if else logic here..

**Comment by Shashwat545** [comment]
> There are few modifcations which I want you to consider:
> 
> 1. The logic behind preprocessing should go in preprocessing function. By this I mean:
>    if else
>    This if else block should go to preprocess function, if there is no change just return original value, that's it. This will enable the function to be reused anywhere.
> 2. Don't hardcode rupee symbol by removing line number 193 from NewBill Activity. This might face difficulty in some devices. Rather use that getString(R.string.rupee, totalAmount) functinon to get rupee symbol.
> 3. Merge your changes as a single commit rather than three commits.
>    Thanks

@sushant-iit I have made the required modifications. Please check if it is fine now or any further changes are needed to be done.

File: app/src/main/java/com/sushant/quickbills/activity/NewBillActivity.kt [hunk 3]
```diff
@@ -190,7 +191,8 @@ class NewBillActivity : AppCompatActivity(), RecyclerParticularsAdapter.OnClickL
                 )
                 particularList.add(newParticularItem)
                 totalAmount += amount
-                bill_amount.text = getString(R.string.rupee, totalAmount)
+                val ans= preprocessDisplayText(totalAmount)
+                bill_amount.text=(getString(R.string.rupee_symbol_only)+" "+ans)
```
Comment (sushant-iit) [inline app/src/main/java/com/sushant/quickbills/activity/NewBillActivity.kt]: ans variable feels unnecessary here:
Make it like this 
bill_amount.text=(getString(R.string.rupee_symbol_only)+" "+preprocessDisplayText(totalAmount))

File: app/src/main/java/com/sushant/quickbills/utils/preprocessDisplayText.kt [hunk 4]
```diff
@@ -0,0 +1,13 @@
...
```
Comment (sushant-iit) [inline app/src/main/java/com/sushant/quickbills/utils/preprocessDisplayText.kt]: Give anss a reasonable name like preprocessedText or something

**Review by sushant-iit — CHANGES_REQUESTED** [review]
And, please remove so many commits from your branch. You should have only one commit. It does not make sense to have 4 commits for this simple issue.

**Comment by Shashwat545** [comment]
Please check now if anything else is to be done

**Review by sushant-iit — APPROVED** [review]
#merged

**Comment by rahulpathak-github** [comment]
@sushant-iit Please add the difficulty tag too.

**Comment by sushant-iit** [comment]
@rahulpathak-github @Shashwat545 Done

**Pull request merged**